src/xcode/ENA/ENA/Source/Scenes/Checkin/QRCodeScanner/CheckinQRCodeScannerViewController.swift

```diff
@@ -96,23 +172,64 @@ class CheckinQRCodeScannerViewController: UIViewController {
 
 		viewModel.deactivateScanning()
 
-		let alert = UIAlertController(title: "QR-Code nicht gültig", message: nil, preferredStyle: .alert)
+		let alert = UIAlertController(
+			title: AppStrings.Checkin.QRScanner.Error.title,
+			message: AppStrings.Checkin.QRScanner.Error.description,
+			preferredStyle: .alert
+		)
 		alert.addAction(
 			UIAlertAction(
-				title: "abbrechen",
+				title: AppStrings.Common.alertActionCancel,
 				style: .cancel,
 				handler: { [weak self] _ in
 					self?.dismiss()
 				}
 			)
 		)
-		alert.addAction(UIAlertAction(title: "OK", style: .default, handler: { [weak self] _ in
+		alert.addAction(UIAlertAction(title: AppStrings.Common.alertActionOk, style: .default, handler: { [weak self] _ in
 			self?.viewModel.activateScanning()
 		}))
 
 		DispatchQueue.main.async { [weak self] in
 			self?.present(alert, animated: true)
 		}
 	}
+	private func setNeedsPreviewMaskUpdate() {
+		guard needsPreviewMaskUpdate else { return }
+		needsPreviewMaskUpdate = true
+
+		DispatchQueue.main.async(execute: updatePreviewMaskIfNeeded)
+	}
 
+	private func updatePreviewMaskIfNeeded() {
+		guard needsPreviewMaskUpdate,
+			  let previewLayer = previewLayer,
+			  focusView.backdropOpacity > 0 else {
```

Comment:
What is this line preventing?

Reply:
The screen is almost identical to the one in exposure Submission, So i used the whole function to create the opacity mask, to be honest I'm not sure  about the guard checks I kept them just in case they handle some special case but when i removed them no change happened.
should i remove them ?

Reply:
If the code does nothing, please remove it.